lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/LeastSquaresExpansion.cxx

```diff
@@ -0,0 +1,195 @@
+}
+
+
+/* String converter */
+String LeastSquaresExpansion::__repr__() const
+{
+  return OSS() << "class=" << getClassName();
+}
+
+
+/* Computes the functional chaos */
+void LeastSquaresExpansion::run()
+{
+  // Check if we have to transform the input sample in order to adapt it
+  // to the basis
+  const Distribution measure(basis_.getMeasure());
+  const Bool noTransformation = (distribution_ == measure);
+  Sample transformedInputSample;
+  DistributionTransformation transformation;
+  if (noTransformation) transformedInputSample = inputSample_;
+  else
+    {
+      transformation = DistributionTransformation(distribution_, basis_.getMeasure());
+      transformedInputSample = transformation(inputSample_);
+    }
+  // Build the relevant functions from the basis
+  Indices indices(basisSize_);
+  indices.fill();
+  FunctionCollection functions(basisSize_);
+  for (UnsignedInteger i = 0; i < basisSize_; ++i)
+    functions[i] = basis_.build(i);
+  // Now build the design proxy and the least-squares method
+  DesignProxy designProxy(transformedInputSample, functions);
+  LeastSquaresMethod leastSquaresMethod = LeastSquaresMethod::Build(methodName_, designProxy, weights_, indices);
+  const UnsignedInteger outputDimension = outputSample_.getDimension();
+  Sample coefficients(basisSize_, outputDimension);
+  const Matrix weightedDesign(leastSquaresMethod.computeWeightedDesign());
+  Point residuals(outputDimension);
+  Point relativeErrors(outputDimension);
+  const UnsignedInteger sampleSize = transformedInputSample.getSize();
+  for (UnsignedInteger j = 0; j < outputDimension; ++j)
+    {
+      const Sample marginalOutputSample(outputSample_.getMarginal(j));
+      const Point rhs(marginalOutputSample.asPoint());
+      const Point coeffsJ(leastSquaresMethod.solve(rhs));
+      for (UnsignedInteger i = 0; i < basisSize_; ++i)
+        coefficients(i, j) = coeffsJ[i];
+      // Now the two errors
+      const Scalar quadraticResidual = (weightedDesign * coeffsJ - rhs).normSquare();
+      residuals[j] = std::sqrt(quadraticResidual) / sampleSize;
+      const Scalar empiricalError = quadraticResidual / sampleSize;
+      relativeErrors[j] = empiricalError / marginalOutputSample.computeVariance()[0];
+    }
+  // Build the result
+  if (noTransformation)
+    {
+      const IdentityFunction identity(distribution_.getDimension());
+      result_ = FunctionalChaosResult(inputSample_, outputSample_, distribution_, identity, identity, basis_, indices, coefficients, functions, residuals, relativeErrors);
+    }
+  else
+    result_ = FunctionalChaosResult(inputSample_, outputSample_, distribution_, transformation, transformation.inverse(), basis_, indices, coefficients, functions, residuals, relativeErrors);
+}
+
+/* Method save() stores the object through the StorageManager */
+void LeastSquaresExpansion::save(Advocate & adv) const
+{
+  FunctionalChaosAlgorithm::save(adv);
+  adv.saveAttribute( "weights_", weights_ );
```

Comment:
Don't we have `weights` at an upper level ? (ie `FunctionalChaos`)

Reply:
yes, fixed